src/OpenLoco/Title.cpp

```diff
@@ -81,13 +161,73 @@ namespace OpenLoco::Title
         initialiseViewports();
         sub_4284C8();
         Gui::init();
-        sub_444357();
+        reset();
         Gfx::invalidateScreen();
         resetScreenAge();
 
         Audio::playTitleScreenMusic();
     }
 
+    // 0x00444387
+    void update()
+    {
+        if (!isTitleMode())
+            return;
+
+        resetScreenAge();
+
+        if (_waitCounter > 0)
+        {
+            _waitCounter -= 1;
+            return;
+        }
+
+        do
+        {
+            auto& command = *_sequenceIterator++;
```

Comment:
Theres one pitfall here. If the list does not contain some entry like ```ResetStep``` this iterator will eventually become invalid and causes a crash. It should probably check that it did not reach the end of the list before doing anything.

Reply:
You're suggesting we employ something like the following?
```suggestion
            // Sanity check
            if (_sequenceIterator >= TitleSequence.end())
                _sequenceIterator = _titleSequence.begin();

            auto& command = *_sequenceIterator++;
```

Reply:
I don't think it should do anything when its not specified, all it has to do is return.

Reply:
Sure, that's likely better, yeah. Like so, then?
```suggestion
            if (_sequenceIterator >= TitleSequence.end())
                return;

            auto& command = *_sequenceIterator++;
```

Reply:
That will do. From this point it will just continue playing the scenario, so not sure if there has to be a graceful exit or something.